### PR TITLE
Bj0rnen/kill lingering repairs

### PR DESCRIPTION
**Pull request by Bj0rnen**
Combat #124 and potentially similar situations by cancelling all repairs on nodes that _shouldn't_ be repairing anything according to Reaper itself.

### CODE REVIEW

**Comment by varjoranta** [comment]
:+1:

File: src/main/java/com/spotify/reaper/service/SegmentRunner.java [hunk 1]
```diff
@@ -152,7 +156,20 @@ private void runRepair() {
       RepairUnit repairUnit = context.storage.getRepairUnit(segment.getRepairUnitId()).get();
       String keyspace = repairUnit.getKeyspaceName();
 
-      if (!canRepair(segment, keyspace, coordinator)) {
+      LazyInitializer<Set<String>> busyHosts = new LazyInitializer<Set<String>>() {
```
Comment (mmatalka) [inline src/main/java/com/spotify/reaper/service/SegmentRunner.java]: Add a comment explaining why it's a `LazyInitializer`.  I think I get why, but it took reading some code to figure it out.

File: src/main/java/com/spotify/reaper/storage/MemoryStorage.java [hunk 2]
```diff
@@ -300,6 +304,31 @@ public boolean updateRepairSegment(RepairSegment newRepairSegment) {
   }
 
   @Override
+  public Collection<RepairParameters> getOngoingRepairsInCluster(String clusterName) {
```
Comment (mmatalka) [inline src/main/java/com/spotify/reaper/storage/MemoryStorage.java]: Is there anyway to make this method feel alittle bit less like being punched in the face with a boilerplate?  It just feels like all the logic is really obscured by  the surrounding verbosity.  A comment describing the algorithm at least?

Comment (Bj0rnen) [inline src/main/java/com/spotify/reaper/storage/MemoryStorage.java]: The nicest way I could come up with sadly was loops. The real answer is hopefully Java8. I think we should go there with the 2.1 support.

**Comment by mmatalka** [comment]
:+1: no idea on correctness, per se, but the code looks good.

**Pull request merged**